lib/driver.js

```diff
@@ -219,6 +233,46 @@ class EspressoDriver extends BaseDriver {
     }
   }
 
+  async onPostConfigureApp ({cachedAppInfo, isUrl, appPath}) {
```

Comment:
noticed still this `appPath` could have `.zip` extension, so `.zip` file always handles as `this.adb.extractUniversalApk` even it is `apk` after unzip.
https://github.com/appium/ruby_lib_core/issues/370

Reply:
https://github.com/appium/appium/pull/16548